aidial_adapter_bedrock/llm/model/anthropic.py

```diff
@@ -145,4 +164,130 @@ async def _apredict(
+        response: httpx.Response,
+        client: AsyncAnthropic,
+    ):
+        super().__init__(cast_to=cast_to, response=response, client=client)
+        self.prompt_tokens = 0
+        self.completion_tokens = 0
+        self.stop_reason: str | None = None
+
+    async def on_stream_event(self, event: MessageStreamEvent):
+        if isinstance(event, MessageStartEvent):
+            self.prompt_tokens = event.message.usage.input_tokens
+        elif isinstance(event, MessageDeltaEvent):
+            self.completion_tokens += event.usage.output_tokens
+            self.stop_reason = event.delta.stop_reason
+
+
+class AnthropicChat(ChatModel):
+    storage: Optional[FileStorage]
+    client: AsyncAnthropicBedrock
+
+    async def achat(
+        self,
+        consumer: Consumer,
+        params: ModelParameters,
+        messages: List[Message],
+    ):
+        if len(messages) == 0:
+            raise ValidationError("List of messages must not be empty")
+
+        prompt, claude_messages = await to_claude_messages(
+            messages, self.storage
+        )
+        completion_params = {
+            "max_tokens": params.max_tokens or DEFAULT_MAX_TOKENS_ANTHROPIC,
+            "stop_sequences": NOT_GIVEN if params.stop is None else params.stop,
+            "system": prompt or NOT_GIVEN,
+            "temperature": NOT_GIVEN
+            if params.temperature is None
+            else params.temperature / 2,
+            "top_p": NOT_GIVEN if params.top_p is None else params.top_p,
+        }
+        if params.stream:
+            await self.invoke_streaming(
+                consumer, claude_messages, completion_params
+            )
+        else:
+            await self.invoke(consumer, claude_messages, completion_params)
+
+    async def invoke_streaming(
+        self,
+        consumer: Consumer,
+        messages: List[MessageParam],
+        params: dict[str, Any],
+    ):
+        log.debug(
+            f"Streaming request: messages={messages}, model={self.model}, params={params}"
+        )
+        async with self.client.messages.stream(
+            messages=messages,
+            model=self.model,
+            event_handler=UsageEventHandler,
+            **params,
+        ) as stream:
+            async for text in stream.text_stream:
+                consumer.append_content(text)
+            consumer.close_content(to_dial_finish_reason(stream.stop_reason))
+
+            consumer.add_usage(
+                TokenUsage(
+                    prompt_tokens=stream.prompt_tokens,
+                    completion_tokens=stream.completion_tokens,
+                )
+            )
+
+    async def invoke(
+        self,
+        consumer: Consumer,
+        messages: List[MessageParam],
+        params: dict[str, Any],
+    ):
+        log.debug(
+            f"Request: messages={messages}, model={self.model}, params={params}"
+        )
+        message = await self.client.messages.create(
```

Comment:
Please add parameter `stream=False`. 
Otherwise, `pyright` seems to unable to derive the type of `message` properly (

Reply:
Done.